kafka-1.0.x/src/main/scala/monix/kafka/CommittableOffset.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.kafka
+
+import monix.eval.Task
+import org.apache.kafka.common.TopicPartition
+
+/** Represents offset for specified topic and partition that can be
+  * committed by [[commit]] method call.
+  * To achieve good performance it is recommended to use batched commit with
+  * [[CommittableOffsetBatch]] class.
+  *
+  * @param topicPartition is the topic and partition identifier
+  *
+  * @param offset is the offset to be committed
+  *
+  * @param commitBatch is the function for batched commit realized as closure
+  *        in [[KafkaConsumerObservable]] context. This decision was made for
+  *        thread-safety reasons.
+  */
+final class CommittableOffset(
+  val topicPartition: TopicPartition,
+  val offset: Long,
+  private[kafka] val commitBatch: Map[TopicPartition, Long] => Task[Unit]) {
```

Comment:
Perhaps we could wrap this function in a `trait`. What do you think?

Reply:
We can do something like this:

```scala
trait CommittableOffset {
  def topicPartition: TopicPartition
  def offset: Long
  def commitBatch(batch: Map[TopicPartition, Long]): Task[Unit]

  final def commit(): Task[Unit] = commitBatch(Map(topicPartition -> offset))
}

object CommittableOffset {

  def apply(
    topicPartition: TopicPartition,
    offset: Long,
    commitBatchCallback: Map[TopicPartition, Long] => Task[Unit]): CommittableOffset =
    new CommittableOffset {
      override val topicPartition: TopicPartition = topicPartition
      override val offset: Long = offset
      override def commitBatch(batch: Map[TopicPartition, Long]): Task[Unit]
 = commitBatchCallback(batch)
    }
}

```

Did you mean exactly that?

Reply:
I've had something like this in mind:
```scala
// hopefully you have idea for better name:)
trait Commit {
  def commitBatch(batch: Map[TopicPartition, Long]): Task[Unit]
  // perhaps other versions, like sync and async
}

final class CommittableOffset(
  val topicPartition: TopicPartition,
  val offset: Long,
  private[kafka] commit: Commit) {
    def commit(): Task[Unit] = commit.commitBatch(Map(topicPartition -> offset))
}
```

Reply:
I introduced this trait and it looks very well!